applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -268,11 +337,17 @@ static bool app_event_handler(const struct app_event_header *aeh)
 				}
 				get_req_modules(&req_modules_bm);
 			} else {
-				enum dvfs_frequency_setting required_freq =
-								check_required_frequency();
-				if (required_freq != current_freq) {
-					set_dvfs_freq(required_freq);
+				int ret = clock_control_get_rate(dvfs_clock_dev,
```

Comment:
Shouldn't we get the clock rate and try to update frequency unconditionally? Currently you seemingly assume that initialization would require high frequency (which might not be true).